packages/astro/src/core/build/generate.ts

```diff
@@ -145,10 +145,12 @@ export async function generatePages(opts: StaticBuildOptions, internals: BuildIn
 		const baseDirectory = getOutputDirectory(opts.settings.config);
 		const renderersEntryUrl = new URL('renderers.mjs', baseDirectory);
 		const renderers = await import(renderersEntryUrl.toString());
+		const { onRequest: middleware } = await import(new URL('middleware.mjs', baseDirectory).toString());
```

Comment:
This is new, can you explain why this is imported this way? I can see that it's passed into `createBuildManifest`, how was the middleware previously imported in generation? I don't see an old import() that was removed.